pkg/api/addtask.go

```diff
@@ -0,0 +1,81 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/wolfalex23/go_final_project/pkg/db"
+)
+
+func checkDate(task *db.Task) error {
+
+	now := time.Now()
+
+	if task.Date == "" {
+		task.Date = now.Format(DateFormat)
+	}
+
+	t, err := time.Parse(DateFormat, task.Date)
+
+	if err != nil {
+		return fmt.Errorf("invalid data format: %v", err)
+	}
+
+	var next string
+
+	if task.Repeat != "" {
+		next, err = NextDate(now, task.Date, task.Repeat)
+		if err != nil {
+			return fmt.Errorf("invalid repeat rule: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
			return fmt.Errorf("invalid repeat rule: %w", err)
```

Reply:
/all